GTMAppAuth/Sources/AuthSession.swift

```diff
@@ -225,8 +225,7 @@ open class AuthSession: NSObject, GTMSessionFetcherAuthorizer, NSSecureCoding {
     serialAuthArgsQueue.sync {
       authorizationArgs.append(args)
     }
-    let additionalRefreshParameters = tokenRefreshDelegate?
-      .additionalRefreshParameters(authSession: self)
+    let additionalRefreshParameters = delegate?.additionalRefreshParameters?(forAuthSession: self)
     let authStateAction = {
       (accessToken: String?, idToken: String?, error: Swift.Error?) in
```

Comment:
Relatedly, let's think about how we handle errors coming from AppAuth here.  It seems that any token refresh error at this point gets mapped to `Error.accessTokenEmptyForRequest`, which is accurate but doesn't provide any of the [error context](https://github.com/google/GoogleSignIn-iOS/blob/625dc9edd205d8c31a1a8037ac6a257a28cd7405/GoogleSignIn/Sources/GIDEMMErrorHandler.m#L67) coming from AppAuth.  Perhaps `NSError`'s `userInfo` and [`NSUnderlyingErrorKey`](https://developer.apple.com/documentation/foundation/nsunderlyingerrorkey) could be an option.

Reply:
Resolving as this is addressed by a subsequent change that retains any error from AppAuth by passing it to the queued arguments:

```
    let authStateAction = {
      (accessToken: String?, idToken: String?, error: Swift.Error?) in
      self.serialAuthArgsQueue.sync { [weak self] in
        guard let self = self else { return }
        for var queuedArgs in self.authorizationArgs {
          if let error = error {
            // Give `queuedArgs` most recent error from AppAuth
            queuedArgs.error = error
          }
          self.authorizeRequestImmediately(args: queuedArgs, accessToken: accessToken)
        }
        self.authorizationArgs.removeAll()
      }
    }
```